pgosupport.cmake

```diff
@@ -0,0 +1,68 @@
+function(clr_pgo_unknown_arch)
+    if (WIN32)
+        message(FATAL_ERROR "Only AMD64, ARM and I386 are supported for PGO")
+    else()
+        message(FATAL_ERROR "PGO not currently supported on the current platform")
+    endif()
+endfunction(clr_pgo_unknown_arch)
+
+# Adds Profile Guided Optimization (PGO) flags to the current target
+function(add_pgo TargetName)
+    if(WIN32)
+        set(ProfileFileName "${TargetName}.pgd")
+    endif(WIN32)
+
+    file(TO_NATIVE_PATH
+        "$ENV{__PackagesDir}/Microsoft.DotNet.OptimizationData.Coreclr/$ENV{__BuildOS}.$ENV{__BuildArch}/${ProfileFileName}"
+        ProfilePath
+    )
+
+    # Enable PGO only for optimized configs
+    set(ConfigTypeList RELEASE RELWITHDEBINFO)
+
+    foreach(ConfigType IN LISTS ConfigTypeList)
```

Comment:
Looping over configs makes no sense on single target cmake generators, like the Unix make. The cmake is run for a single specific configuration.
You can use generator expressions instead of looping over the configs. That would work in both cases. For example, the first set_property would be
```cmake
set_property(TARGET ${TargetName} APPEND_STRING PROPERTY LINK_FLAGS $<$<OR:$<CONFIG:Release>,$<CONFIG:Relwithdebinfo>>:/LTCG /GENPROFILE>
```